### ISSUE TITLE
Feature request: support HiDPI display / editor display scale setting

### ISSUE THREAD

**Issue by kezzbracey** [Orama-Interactive/Pixelorama#140]
Hi, thanks so much for the amazing work with Pixelorama!

It would be brilliant if there could be support for HiDPI displays as at the moment the interface is quite small in scale and a bit hard on the eyes:

<img width="1920" alt="pixelorama" src="https://user-images.githubusercontent.com/5533167/73147462-a5c7c580-410b-11ea-9bf9-ec4960d49315.png">

Godot has a "Display Scale" setting for the editor that would solve the problem beautifully if it were possible to bring it across:

![godotscaling](https://user-images.githubusercontent.com/5533167/73147475-b0825a80-410b-11ea-945e-9084cafc6a76.jpg)

Thanks for considering, and for this great project. :)

**Comment by OverloadedOrama** [comment]
Hello, thanks for the kind words and the suggestion! We have tried scaling the UI to support HiDPI displays, but for some reason the mouse coordinates in the canvas get messed up. I believe it is related to [this Godot issue](https://github.com/godotengine/godot/issues/34805). I'm not sure how to fix it yet but I will try. I might need some help with this one though if anyone has any ideas!

**Comment by Calinou** [comment]
Adding proper hiDPI support for a non-game application is unfortunately not possible given what Godot currently exposes. This could be achieved once https://github.com/godotengine/godot/pull/21446 is merged.

**Comment by OverloadedOrama** [comment]
That's a bummer. So I guess we'll have to wait for Godot 4.0 to come out to solve the issue, since your PR to Godot belongs to the 4.0 milestone. Unless it's possible for it to be merged at a version like 3.2.1?

**Comment by Calinou** [comment]
@OverloadedOrama Since it changes the function signature in a backwards-compatible manner, it might be able to be merged into 3.2.1, but no guarantees.

**Comment by Renduin** [comment]
I'd love to see this as well. The app looks really great, but it's a bit too hard on the eyes on my 4k monitor. I'll keep an eye on it for future projects!

**Comment by adojck** [comment]
I think this feature is very important as HiDPI displays are becoming very common. Currently it's pretty hard to use Pixelorama because of tiny UI on HiDPI screens. Hope to see this support soon!

**Comment by Calinou** [comment]
@adojck I think the previous workaround could likely be fixed to not break mouse coordinates, which should make it usable again. See https://github.com/godotengine/godot/issues/14542#issuecomment-636306369, except that we'll scale by a fixed factor (2.0) instead of a factor that varies depending on the window size.
